resource/discovery.go

```diff
@@ -12,10 +12,11 @@ type (
 	DiscoveryQuery struct {
 		API   API
 		Model Model
+		Extra map[string]interface{}
 	}
 
 	// DiscoveryFunc is a function that discovers component configurations.
-	DiscoveryFunc func(ctx context.Context, logger logging.Logger) (interface{}, error)
+	DiscoveryFunc func(ctx context.Context, logger logging.Logger, extra map[string]interface{}) (interface{}, error)
```

Comment:
For a split-second I was worried this might break some modular component's discovery function, but then I remembered this wasn't possible until recently :slightly_smiling_face:

Reply:
True! Will still announce a breaking change incase